L1Trigger/L1THGCal/src/veryfrontend/HGCalVFECompressionImpl.cc

```diff
@@ -1,5 +1,87 @@
 #include "L1Trigger/L1THGCal/interface/veryfrontend/HGCalVFECompressionImpl.h"
 
 HGCalVFECompressionImpl::
-HGCalVFECompressionImpl(const edm::ParameterSet& conf)
-{}
+HGCalVFECompressionImpl(const edm::ParameterSet& conf):
+    exponentBits_(conf.getParameter<uint32_t>("exponentBits")),
+    mantissaBits_(conf.getParameter<uint32_t>("mantissaBits")),
+    rounding_(conf.getParameter<bool>("rounding"))
+{
+  saturable_ = false;
+  saturationValue_ = 0xffffffff;
+  if (((1 << exponentBits_) + mantissaBits_ - 1) < 32) {
```

Comment:
Instead of that, it would be better to throw an exception if the coding cannot be done on 32 bits.

Reply:
Ok so let's say for 6-bit exponent and just 2-bit mantissa. The codepoint of 0x7b would give 3758096384, hence the code for 0x7c would exceed the 32-bit space maximum at 4294967295. So should immediately throw and exception in this case, and not considering this at all if the whole codespace cannot map within the 32-bit valuespace?

Reply:
Yes. I think this is the best thing to do to avoid unexpected behaviour.